src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/DefaultVisualStudioWorkspaceAccessor.cs

```diff
@@ -1,28 +1,77 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.ComponentModel.Composition;
-using Microsoft.CodeAnalysis;
 using Microsoft.VisualStudio.Editor.Razor;
 using Microsoft.VisualStudio.Text;
 using MonoDevelop.Ide.TypeSystem;
+using MonoDevelop.Projects;
+using Workspace = Microsoft.CodeAnalysis.Workspace;
 
 namespace Microsoft.VisualStudio.Mac.LanguageServices.Razor
 {
     [System.Composition.Shared]
     [Export(typeof(VisualStudioWorkspaceAccessor))]
     internal class DefaultVisualStudioWorkspaceAccessor : VisualStudioWorkspaceAccessor
     {
-        public DefaultVisualStudioWorkspaceAccessor()
+        private readonly TextBufferProjectService _projectService;
+
+        [ImportingConstructor]
+        public DefaultVisualStudioWorkspaceAccessor(TextBufferProjectService projectService)
         {
-            Workspace = TypeSystemService.Workspace;
+            if (projectService == null)
+            {
+                throw new ArgumentNullException(nameof(projectService));
+            }
+
+            _projectService = projectService;
         }
 
-        public override Workspace Workspace { get; }
+        public override Workspace Workspace => TypeSystemService.Workspace;
```

Comment:
Just for clarity, I'm assuming you plan to delete this property once you remove all of the uses of it right?

Reply:
Yup, that's what [this](https://github.com/aspnet/Razor/issues/2001) workitem tracks.